### PR TITLE
refactor oshinko-deploy to be more relevant

### PR DESCRIPTION
**Pull request by elmiko**
This refactors the deploy script to use the most up-to-date methods for
deploying the oshinko suite.

Changes
* refactor deploy script
* update readmes

### CODE REVIEW

File: tools/oshinko-deploy.sh [hunk 1]
```diff
@@ -55,71 +82,54 @@ while getopts :s:w:p:u:h opt; do
...
+if $OS_ALLINONE
 then
-    docker pull $SPARK_IMAGE
+    oc cluster up
 fi
-docker tag $SPARK_IMAGE $REGISTRY/$PROJECT/oshinko-spark
-docker push $REGISTRY/$PROJECT/oshinko-spark
 
-# set up the oshinko service account
+oc login $OS_CLUSTER -u $OS_USER
```
Comment (tmckayus) [inline tools/oshinko-deploy.sh]: Are the OS_ALLINONE case and the OS_CLUSTER case mutually exclusive?  If I'm using oc cluster up, then I've got no use for OS_CLUSTER, correct? Should we stop them from using -d and -c at the same time?

File: tools/oshinko-deploy.sh [hunk 2]
```diff
@@ -1,51 +1,78 @@
...
 #
-# Usage:
-# $ oshinko-deploy.sh {route IP} {project name}
+# In either case, access to the `oc` command is required.
 #
-# route IP -- address to use in the exposed route information
-# project name -- project to deploy oshinko into
+# example usage:
+#
+#    $ oshinko-deploy.sh -c 10.0.1.100 -w oshinko.10.0.1.100.xip.io
```
Comment (tmckayus) [inline tools/oshinko-deploy.sh]: -w flag changed, this should be a -o

**Comment by tmckayus** [comment]
I tried the -d option, worked if I ran it as root.  I also tried adding my user to the docker group and restarting docker but I was still unable to run oshinko-deploy -d as a regular user (oc cluster up fails)

File: tools/oshinko-deploy.sh [hunk 3]
```diff
@@ -55,71 +81,66 @@ while getopts :s:w:p:u:h opt; do
...
+    fi
 fi
-docker tag $SPARK_IMAGE $REGISTRY/$PROJECT/oshinko-spark
-docker push $REGISTRY/$PROJECT/oshinko-spark
 
-# set up the oshinko service account
+oc login $OS_CLUSTER -u $OS_USER
+
+oc new-project $PROJECT
+
```
Comment (elmiko) [inline tools/oshinko-deploy.sh]: i think all the oc commands specify the `-n` option to set the project, i thought adding an `oc project $PROJECT` might be redundant, but i can add for sure.

you're probably right about the user thing, i'll add more rejection stuff if you specify a user and all-in-one

Comment (tmckayus) [inline tools/oshinko-deploy.sh]: ah, it must be the new-app that's failing then

Error from server: project "myproject" already exists
serviceaccount "oshinko" created
template "oshinko" created
error: no match for "oshinko"

The 'oc new-app' command will match arguments to the following types:

Comment (tmckayus) [inline tools/oshinko-deploy.sh]: yep, you were right, -n on the new-app will fix it, no need for "oc project $PROJECT"

**Pull request merged**